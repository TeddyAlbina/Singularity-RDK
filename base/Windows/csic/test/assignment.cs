public class test {
	public static void Main() {
		int a, b, c, d, e = 1;
		a = (b = (c = (d = 2)));
		System.Console.WriteLine(a);
	}
}
