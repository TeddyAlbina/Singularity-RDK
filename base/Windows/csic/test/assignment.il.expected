.assembly 'assignment' { }
.assembly extern 'mscorlib' {
.ver 1:0:5000:0
.publickeytoken = ( b7 7a 5c 56 19 34 e0 89 )
}
.class public 'test'
	extends [mscorlib]System.Object
{
.method hidebysig public static void 'Main'() {
.entrypoint
.locals init ([0]int32 't0')
.locals init ([1]int32 't1')
.locals init ([2]int32 't2')
.locals init ([3]int32 't3')
.locals init ([4]int32 't4')
ldc.i4 1
stloc 4 // e
ldc.i4 2
stloc 3 // d
ldc.i4 2
stloc 2 // c
ldc.i4 2
stloc 1 // b
ldc.i4 2
stloc 0 // a
ldloc 0 // a
call  void [mscorlib]System.Console::'WriteLine'(int32)
ret
.maxstack 1
} // end of method test.Main
.method hidebysig specialname rtspecialname public void .ctor() {
ldarg.0
call instance void [mscorlib]System.Object::.ctor()
ret
.maxstack 1
} // end of constructor test.test
} // end of class test
